### ISSUE TITLE
VPC Interface Endpoint is not always auto accepted

### ISSUE THREAD

**Issue by michal-malec** [terraform-providers/terraform-provider-aws#13387]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v0.12.24

### Affected Resource(s)

* aws_vpc_endpoint

### Terraform Configuration Files

```hcl
resource "aws_vpc_endpoint_service" "nlb" {
  acceptance_required        = "true"
  network_load_balancer_arns = var.NETWORK_LOAD_BALANCERS
  allowed_principals         = var.ALLOWED_PRINCIPALS

  tags = local.default_tags
}

resource "aws_vpc_endpoint" "interface" {
  vpc_id              = var.VPC_ID
  vpc_endpoint_type   = "Interface"
  service_name        = aws_vpc_endpoint_service.nlb.service_name
  auto_accept         = "true"
  security_group_ids  = [aws_security_group.endpoint_security_group[0].id]
  subnet_ids          = var.SUBNET_IDS
  private_dns_enabled = "false"

  tags = local.default_tags
}
```

### Expected Behavior

VPC Endpoint is automatically accepted and available.

### Actual Behavior

Sometimes VPC Interface Endpoint connected to VPC Endpoint Service is not accepted automatically and it stays in "pending acceptance" status. There is no indication in terraform that auto acceptance failed.

### Steps to Reproduce

1. Run `terraform apply`
2. Open AWS Console and check whether VPC Endpoint was accepted

_This issue occurs only from time to time._

**Comment by ewbankkit** [comment]
@michal-malec Thanks for raising this issue.
I see the problem

https://github.com/terraform-providers/terraform-provider-aws/blob/fcabc0f8ac265ab234f7ec8bc9056b1fdec8938e/aws/resource_aws_vpc_endpoint.go#L426-L429

`pendingAcceptance` is being handled as a valid _Target_ state when waiting for an endpoint to become active. In the case where `auto_accept = true`, this is not correct and `pendingAcceptance` should be an additional _Pending_ state.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.68.0 of the Terraform AWS Provider, later this week. Thanks to @ewbankkit for the implementation. 👍
